Remove bower

chfsx opened the pull request:
Isn't it true that we once moved all JS dependencies from bower to npm? but in the process, the bower modules were never removed.

chfsx committed:
Remove bower

mjansenDatabay commented:
There is still a usage in:

* `\ilLTIConsumerContentGUI::showEmbedded`
* `\ilPreviewGUI::initPreview`
* `\ilYuiUtil::YUI_BASE`
* `\ilAccordionGUI::$owl_path`

chfsx commented:
@mjansenDatabay Thanks for looking into this, I think the following usages should be moved to npm as well or maybe they can be dropped anyway: 

- \ilLTIConsumerContentGUI::showEmbedded
- \ilYuiUtil::YUI_BASE
- \ilAccordionGUI::$owl_path

For \ilPreviewGUI::initPreview: this will be removed in ILIAS 9 anyways (and already is in trunk)

chfsx committed:
Moved YUI to npm and fixed some other paths

chfsx commented:
I now moved YUI2 to npm as well and fixed some other paths, see https://github.com/ILIAS-eLearning/ILIAS/pull/5687/commits/b46373fc4ad7bc3de43f98daf720125f175f28c2

chfsx committed:
Fixed test :-)

alex40724 commented:
I can take care of \ilAccordionGUI::$owl_path in the trunk.

matthiaskunkel commented:
Jour Fixe, 20 MAR 2023: We highyl appreciate this suggestion and accept the PR for trunk.

chfsx commented on Services/Accordion/classes/class.ilAccordionGUI.php:
```diff
@@ -44,7 +44,7 @@ class ilAccordionGUI
     protected string $containerclass = "";
     protected string $id = "";
     protected bool $head_class_set = false;
-    public static string $owl_path = "./libs/bower/bower_components/owl.carousel/dist";
```
@alex40724 FYI

alex40724 replied:
@chfsx Great! Thanks.

chfsx commented on Modules/LTIConsumer/classes/class.ilLTIConsumerContentGUI.php:
```diff
@@ -313,7 +313,7 @@ protected function showEmbedded(): void
 
             $v = DEVMODE ? '?vers=' . time() : '?vers=' . ILIAS_VERSION_NUMERIC;
             $tpl->setVariable("DELOS_CSS_HREF", 'templates/default/delos.css' . $v);
-            $tpl->setVariable("JQUERY_SRC", 'libs/bower/bower_components/jquery/dist/jquery.js' . $v);
```
@Uwe-Kohnle FYI
